libs/api/ebpf_api.cpp

```diff
@@ -2084,47 +2086,77 @@ _ebpf_pe_add_section(
         EBPF_LOG_EXIT();
         return 0;
     }
+
     std::string elf_section_name = pe_context->section_names[pe_section_name];
     std::string program_name = pe_context->program_names[pe_section_name];
 
     ebpf_section_info_t* info = (ebpf_section_info_t*)ebpf_allocate(sizeof(*info));
     if (info == nullptr) {
         pe_context->result = EBPF_NO_MEMORY;
-        EBPF_LOG_EXIT();
-        return 1;
+        return_value = 1;
+        goto Exit;
     }
 
     memset(info, 0, sizeof(*info));
-    info->section_name = _strdup(elf_section_name.c_str());
-    info->program_name = _strdup(program_name.c_str());
+    info->section_name = ebpf_duplicate_string(elf_section_name.c_str());
+    if (info->section_name == nullptr) {
+        pe_context->result = EBPF_NO_MEMORY;
+        return_value = 1;
+        goto Exit;
+    }
+
+    info->program_name = ebpf_duplicate_string(program_name.c_str());
+    if (info->program_name == nullptr) {
+        pe_context->result = EBPF_NO_MEMORY;
+        return_value = 1;
+        goto Exit;
+    }
+
     info->program_type = pe_context->section_program_types[pe_section_name];
     info->expected_attach_type = pe_context->section_attach_types[pe_section_name];
+
     info->program_type_name = ebpf_get_program_type_name(&pe_context->section_program_types[pe_section_name]);
     if (info->program_type_name == nullptr) {
         pe_context->result = EBPF_NO_MEMORY;
-        EBPF_LOG_EXIT();
-        return 1;
+        return_value = 1;
+        goto Exit;
+    }
+
+    info->program_type_name = ebpf_duplicate_string(info->program_type_name);
+    if (info->program_type_name == nullptr) {
+        pe_context->result = EBPF_NO_MEMORY;
+        return_value = 1;
+        goto Exit;
```

Comment:
There is a possibility of freeing a const string if `ebpf_duplicate_string` at line 2125 fails. I think we should save the result of `ebpf_get_program_type_name` (line 2118) in a local variable and then duplicate it in line 2125?